src/modules/library.js

```diff
@@ -0,0 +1,199 @@
+let toDoList = [];
+let isEditing = false;
+let todoEdit = null;
+
+const desc = document.getElementById('addToDo');
+
+const saveData = () => {
+  localStorage.setItem('listToDo', JSON.stringify(toDoList));
+};
+
+const getData = () => {
+  const localFormData = JSON.parse(localStorage.getItem('listToDo'));
+  if (localFormData == null) {
+    toDoList = [];
+  } else {
+    toDoList = localFormData;
+  }
+};
+
+const editList = (todo) => {
+  isEditing = true;
+  todoEdit = todo;
+  const desc = document.getElementById('addToDo');
+  desc.value = todo.description;
+  desc.focus();
+};
+
+// MAIN OBJECT
+const displayToDo = () => {
+  const itemsList = document.getElementById('detailedList');
+  itemsList.innerHTML = '';
+
+  // remove a To Do item
+  const removeList = (indexID) => {
+    toDoList = toDoList.filter((ind) => ind.index !== indexID);
+    toDoList = toDoList.map((todo, index) => ({
+      description: todo.description,
+      completed: todo.completed,
+      index: index + 1,
+    }));
+    isEditing = false;
+    desc.value = null;
+    displayToDo();
+  };
+
+  const toggleToDoStatus = (todo) => {
+    for (let i = 0; i < toDoList.length; i += 1) {
+      if (toDoList[i].index === todo.index) {
+        toDoList[i].completed = !todo.completed;
+        break;
+      }
+    }
+    saveData();
+  };
+
+  // Dynamic generation of the To-Do List
+  toDoList.forEach((data) => {
+    const todoLiElement = document.createElement('li');
+    todoLiElement.setAttribute('draggable', true);
+    todoLiElement.classList.add('item');
+    const todoCheckboxElement = document.createElement('input');
+    todoCheckboxElement.classList.add('checkInput');
+    todoCheckboxElement.setAttribute('type', 'checkbox');
+    todoCheckboxElement.setAttribute('name', 'checkbox');
+    todoCheckboxElement.setAttribute('value', data.index);
+
+    const todoDescriptionElement = document.createElement('p');
+    todoDescriptionElement.classList.add('label');
+    todoDescriptionElement.innerText = data.description;
+
+    const actionBtns = document.createElement('div');
+
+    const deleteBtn = document.createElement('button');
+    deleteBtn.classList.add('hide');
+    deleteBtn.setAttribute('type', 'button');
+    deleteBtn.innerHTML = '<i class="fa fa-lg fa-trash">';
+
+    const pointsBtn = document.createElement('button');
+    pointsBtn.classList.add('more-btn');
+    pointsBtn.setAttribute('type', 'button');
+    pointsBtn.innerHTML = '<i class="fa fa-ellipsis-v pointsBtn"></i>';
+
+    todoLiElement.appendChild(todoCheckboxElement);
+    todoLiElement.appendChild(todoDescriptionElement);
+
+    // actionBtns.appendChild(editBtn);
```

Comment:
- [ ] **OPTIONAL CHANGES**

I would suggest Unused pieces of code should be deleted before the submission. it will make your repo more professional.

Reply:
Ok. I fix it !

Reply:
Thanks !